src/main/java/org/fregelang/plugin/idea/run/FregeCommandLineState.java

```diff
@@ -0,0 +1,213 @@
+package org.fregelang.plugin.idea.run;
+
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.CommandLineState;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.process.KillableColoredProcessHandler;
+import com.intellij.execution.process.ProcessHandler;
+import com.intellij.execution.runners.ExecutionEnvironment;
+import com.intellij.ide.highlighter.JavaClassFileType;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.*;
+import com.intellij.openapi.roots.libraries.Library;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.search.FileTypeIndex;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.util.indexing.FileBasedIndex;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+
+/**
+ * Represents a Command that gets executed for compiling and executing Frege Code
+ *
+ * @author rahulsom
+ */
+public class FregeCommandLineState extends CommandLineState {
+  private FregeRunConfiguration runConfiguration;
+
+  public FregeCommandLineState(FregeRunConfiguration runConfiguration, ExecutionEnvironment executionEnvironment) {
+    super(executionEnvironment);
+    this.runConfiguration = runConfiguration;
+  }
+
+  @NotNull
+  @Override
+  protected ProcessHandler startProcess() throws ExecutionException {
+    System.out.println("StartProcess");
+
+    final Project project = runConfiguration.getProject();
+    final ProjectRootManager projectRootManager = ProjectRootManager.getInstance(project);
+
+    final VirtualFile[] sourceRoots = projectRootManager.getContentSourceRoots();
+    final String sourcePath = getSourcePath(sourceRoots);
+    final String fregeClassPath = getClassPathSource();
+    final String projectOutPath = project.getBasePath() + "/out";
+    final String effectiveClassPath = fregeClassPath + ":" + projectOutPath;
+
+    /*
+     * Cross compile from frege to java
+     */
+    ProcessHandler fregec = compile(sourcePath, ".fr", file -> new String[]{
+        "java", "-Xss1m", "-jar", fregeClassPath,
+        "-d", projectOutPath, "-sp", sourcePath, "-nocp", "-greek", "-j", file
+    });
+    if (fregec != null) return fregec;
+
+    /*
+     * Compile Java to bytecode
+     */
+    ProcessHandler javac = compile(projectOutPath, ".java", file -> new String[]{
```

Comment:
In fact I was hoping to just call to the frege compiler and let it compile java code, but it doesn't seem too happy about it. It seems to not find itself in the classpath.

```
calling: javac -cp /Users/rahul/.sdkman/candidates/groovy/current/lib/groovy-2.4.5.jar:../out -d ../out -sourcepath . -encoding UTF-8 ../out/Fib.java 
../out/Fib.java:10: error: package frege does not exist
import frege.Prelude;
            ^
../out/Fib.java:11: error: package frege.control does not exist
import frege.control.Category;
                    ^
../out/Fib.java:12: error: package frege.control does not exist
import frege.control.Semigroupoid;
                    ^

... more of similar stuff ...
```



I'll take a look at Eclipse java compiler and see how we can use it.